plugins/lookup/hashi_vault.py

```diff
@@ -428,6 +444,18 @@ def auth_aws_iam_login(self):
             Display().warning("HVAC should be updated to version 0.9.3 or higher. Deprecated method 'auth_aws_iam' will be used.")
             self.client.auth_aws_iam(**params)
 
+    def auth_jwt(self):
+        params = self.get_options('role_id', 'jwt', 'mount_point')
+        params['role'] = params.pop('role_id')
+        if self.hvac_has_auth_methods and hasattr(self.client.auth.jwt, 'jwt_login'):
+            self.client.auth.jwt.jwt_login(**params)
+        else:
+            # The GCP (and Azure, Kubernetes) auth methods just implement JWT
+            Display().warning("HVAC should be updated to version 0.10.5 or higher. Deprecated method 'auth_gcp' will be used.")
+            if not params['mount_point']:
+                params['mount_point'] = 'jwt'
+            self.client.auth_gcp(**params)
```

Comment:
In my opinion, we shouldn't do this fallback in the code. Instead, I believe we should also enable GCP (and OIDC, Kubernetes and whatever else uses JWT) in the plugin as first-class auth methods, regardless of how they're implemented in HVAC (as we'd call HVAC's implementation), and if the HVAC version available on the system doesn't have JWT, we should fail, suggesting the possible fallback of using GCP or one of the others. We can mention that in docs too.

Also, you're setting the mount point but from what I saw in the HVAC docs, the JWT/OIDC auth methods are the same but differ by their default mount point, so I think we should not set it here; if it's not set in the plugin invocation, it should use the default that HVAC provides.

Reply:
Regarding fallback, I just tried to follow what has been done for the other auth methods. Do you think we should just fail if the installed hvac version doesn't have the `jwt_login` method? I agree on your comment about setting the mount point, and I will remove it - if we decide to keep the fallback. If we remove the fallback, it will go away anyway...

I also agree that we could add first-class support for the other JWT-based auth methods, but I think that should be in another PR.

Reply:
Ah I see what you mean. Short story: The other auth methods "fall back" to what is ultimately the same auth method, it's just that they were moved to a new place within the HVAC library (the `.auth` member) at various points depending on the method in question. 

Long version: Tt was a fallback for auth methods that this plugin mainly already supported, albeit using method calls that were deprecated (really the calls themselves were not, just the way they were being called). In #23 if I updated them all to use the new methods only, then it would be a breaking change as anyone successfully using the plugin with an older version of HVAC would suddenly be broken, but if I did nothing, this plugin would eventually stop working when users updated HVAC to a version where the old method names were removed.

> Do you think we should just fail if the installed hvac version doesn't have the jwt_login method

I think so. It's not unreasonable to expect a specific version of HVAC for specific functionality, as it's already required for the rest of the plugin and decently accounted for the in the documentation.

> I also agree that we could add first-class support for the other JWT-based auth methods, but I think that should be in another PR.

Sure, that's ok too. Even if we don't have a "manual" fallback option for older HVAC versions, I think that's preferable to implementing one. If we add one, we end up having to support it for a quite a while.

I might recommend at least implementing OIDC in this PR too since they are literally the same except in name, and since HVAC handles the only difference (default mount point), the additional code here should be a few lines of copy/paste. But still if you decide not to that's alright, can be done later.

> I agree on your comment about setting the mount point, and I will remove it - if we decide to keep the fallback. If we remove the fallback, it will go away anyway...
d'oh, you're completely right I misread that situation; I see what you're doing there. You do need to override the default mount point if doing fallback, so it doesn't use the default mount for GCP, makes total sense.

Reply:
@briantist I have now added support for OIDC as well. And fail is hvac version is too old. As you recommended/suggested.